### PR TITLE
Partner platform/component pull request

### PR DESCRIPTION
**Pull request by betzw**
Prepare handover to @bcostm

Substitutes #24 and adjusts request from @0xc0170 and @niklas-arm regarding lp_ticker implementation, version number handling etc.

Group of related PRs:
- https://github.com/ARMmbed/mbed-hal-st-stm32f4/pull/25
- https://github.com/ARMmbed/cmsis-core-stm32f4/pull/18
- https://github.com/ARMmbed/mbed-hal-st-stm32f401re/pull/4
- https://github.com/ARMmbed/cmsis-core-nucleo-f401re/pull/4
- https://github.com/ARMmbed/target-st-nucleo-f401re-gcc/pull/4

### CODE REVIEW

**Comment by 0xc0170** [comment]
How did asynch serial get in? :( I see there merging the outstanding PR. Please don't.

**Comment by betzw** [comment]
What should I do?

**Comment by 0xc0170** [comment]
Revert the last commit (merge from asynch_serial), will require push force, be careful.

I'll try to do it locally to test nucleo-f401

**Comment by betzw** [comment]
Should be done.

File: module.json [hunk 1]
```diff
@@ -27,6 +27,9 @@
     "mbed-hal": "*"
   },
   "targetDependencies": {
+    "stm32f401re": {
+      "mbed-hal-st-stm32f401re": "~0.0.1"
```
Comment (0xc0170) [inline module.json]: Same as for cmsis, v0.1.0, @bcostm please publish the latest one.

**Pull request merged**